### PR TITLE
chore(react-storage): refactor Title (composable, hook)

### PR DESCRIPTION
**Pull request by joon-won**
#### Description of changes

- Add Title composable component and corresponding control (TitleControl), hook (useTitle)
- Add corresponding unit tests for each of them

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useTitle.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  it('returns Title data', () => {
+    mockUseControlsContext.mockReturnValue({ data });
+
+    expect(useTitle()).toStrictEqual({
+      title: 'ShinyNewTitle',
+    });
+  });
+
+  it('returns empty string for children if title is missing from data', () => {
```
Comment (cshfang) [inline packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useTitle.spec.tsx:25]: The test says it returns empty string but actually returns undefined. The hook is also not returning children 😛

**Review by cshfang — APPROVED** [review]
Minor comment. Just need to update that test description and this lgtm

File: packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useTitle.spec.tsx (hunk 1 above)
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useTitle.spec.tsx:25]:
```suggestion
  it('returns an undefined value for title when missing from data', () => {
```

**Review by ashwinkumar6 — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useTitle.spec.tsx (hunk 1 above)
Comment (joon-won) [reply packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useTitle.spec.tsx]: Updated the test description, thanks!

**Review by ashwinkumar6 — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**